### ISSUE TITLE
bits/memory_resource.h is missing in gcc.stl.headers.imp

### ISSUE THREAD

**Issue by skypjack** [include-what-you-use/include-what-you-use#1486]
I think `gcc.stl.headers.imp` should include the following:
```cpp
{ include: ["<bits/memory_resource.h>", private, "<memory_resource>", public ] },
```
This is what we get otherwise from IWYU:

> <filename> should add these lines:
> #include <bits/memory_resource.h>              // for polymorphic_allocator

See [here](https://github.com/skypjack/entt/actions/runs/8295051177/job/22701286876#step:5:447) for more details. This is a project of mine where I use IWYU. It has a few false positives, including `bits/memory_resource.h`.
I can also prepare a PR if needed. I would just like confirmation before proceeding because I'm not 100% sure. 🙂

**Comment by kimgr** [comment]
Makes sense. The mappings we use for gnu headers are from the 4.6 era, and we add new ones as they are discovered. If you open a PR, please also add a record in the corresponding map in iwyu_include_picker.cc.

Thanks!

**Comment by skypjack** [comment]
Sure thing. I'll do IT. Thanks for the quick turnaround. 👍

**Comment by skypjack** [comment]
Here we go with #1488

**Comment by skypjack** [comment]
Same thing [here](https://github.com/skypjack/entt/actions/runs/8314254276/job/22751212347#step:5:677) for `bits/uses_allocator.h`, ie `allocator_arg_t` as wrongly reported by IWYU in the linked project:
```cpp
> #include <bits/uses_allocator.h>               // for allocator_arg_t
```
I've added the required fix to #1488:
```cpp
 { include: ["<bits/uses_allocator.h>", private, "<memory>", public ] },
```
`iwyu_include_picker.cc` was updated as well - I hope it's fine. 👍

**Comment by kimgr** [comment]
Fixed on master as of 78472a725b1dd414e0c6ccdef464a8058593d6eb via #1488.
